include/superior_mysqlpp/types/array.hpp

```diff
@@ -170,7 +197,7 @@ namespace SuperiorMySqlpp
 
         constexpr auto maxSize() const
         {
-            return array.max_size();
+            return array.max_size() - int(TerminatingZero);
         }
 
         bool empty() const
```

Comment:
I might ask a lot, but can you add
```suggestion
        constexpr bool empty() const
```
here?

Reply:
Many other methods are constexpr, so why not.

However, it's not that useful for two reasons:
- ideally, it should be added everywhere consistently to be useful, notably ArrayBase constructors.  If you'll do this you'll notice that Debian Jessie version of GCC has imperfect support for C++14 constexpr feature extensions and will reject it
- secondly, one has to wonder if there is any practical usage for this particular type in compile-time at all, as it's designed for database I/O

Reply:
I just assumed, since it's already in `count()` method, it shouldn't hurt here as well. ^_^